pkg/build/builder/dockerutil.go

```diff
@@ -114,3 +119,72 @@ func buildImage(client DockerClient, dir string, dockerfilePath string, noCache
 	}
 	return client.BuildImage(opts)
 }
+
+// tagImage uses the dockerClient to tag a Docker image with name. It is a
+// helper to facilitate the usage of dockerClient.TagImage, because the former
+// requires the name to be split into more explicit parts.
+func tagImage(dockerClient DockerClient, image, name string) error {
+	repo, tag := parsers.ParseRepositoryTag(name)
+	return dockerClient.TagImage(image, docker.TagImageOptions{
+		Repo: repo,
+		Tag:  tag,
+	})
+}
+
+// dockerRun mimics the 'docker run' command. It uses the Docker Remote API to
+// create and start a container and stream its logs to stdout and stderr. The
+// container is removed after it terminates.
+func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, logsOpts docker.LogsOptions) error {
+	// Create a new container.
+	glog.V(4).Infof("Creating container with options {Name:%q Config:%+v HostConfig:%+v} ...", createOpts.Name, createOpts.Config, createOpts.HostConfig)
+	c, err := client.CreateContainer(createOpts)
+	if err != nil {
+		return err
+	}
+
+	containerName := containerNameOrID(c)
+
+	// Container was created, so we defer its removal.
+	defer func() {
+		glog.V(4).Infof("Removing container %q ...", containerName)
+		if err := client.RemoveContainer(docker.RemoveContainerOptions{
+			ID: c.ID,
+		}); err != nil {
+			glog.Warningf("Failed to remove container %q: %v", containerName, err)
+		} else {
```

Comment:
Since the `if` is splitted into multi-line (which really does not help reading this), i prefer not to use 'else' when really not needed. One less condition :-) So personal preference.